openstack/resource_openstack_lb_member_v2.go

```diff
@@ -240,13 +258,28 @@ func resourceMemberV2Delete(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Error creating OpenStack networking client: %s", err)
 	}
 
-	// cache for the parent Load Balancer ID
-	lbID := ""
+	// Get a clean copy of the parent pool.
 	poolID := d.Get("pool_id").(string)
+	parentPool, err := pools.Get(lbClient, poolID).Extract()
+	if err != nil {
+		return fmt.Errorf("Unable to retrieve parent pool %s: %s", poolID, err)
+	}
+
+	// Get a clean copy of the member.
+	member, err := pools.GetMember(lbClient, poolID, d.Id()).Extract()
+	if err != nil {
+		return fmt.Errorf("Unable to retrieve member: %s: %s", d.Id(), err)
+	}
+
+	// Wait for parent pool to become active before continuing.
 	timeout := d.Timeout(schema.TimeoutDelete)
+	err = waitForLBV2Pool(lbClient, parentPool, "ACTIVE", lbPendingStatuses, timeout)
+	if err != nil {
+		return err
+	}
 
-	// Wait for parent pool to become active before continuing
-	err = waitForLBV2Pool(lbClient, poolID, &lbID, "ACTIVE", lbPendingStatuses, timeout)
+	// Wait for the member to become active before continuing.
+	err = waitForLBV2Member(lbClient, parentPool, member, "ACTIVE", lbPendingStatuses, timeout)
```

Comment:
I just realized that we don't need to wait for a member status here. It means that we cannot delete it if it is in an error state.